docs/design/mono/webcil.md

```diff
@@ -0,0 +1,110 @@
+# WebCIL assembly format
+
+## Version
+
+This is version 0 of the Webcil format.
+
+## Motivation
+
+When deploying the .NET runtime to the browser using WebAssembly, we have received some reports from
+customers that certain users are unable to use their apps because firewalls and anti-virus software
+may prevent browsers from downloading or caching assemblies with a .DLL extension and PE contents.
+
+This document defines a new container format for ECMA-335 assemblies
+that uses the `.webcil` extension and uses a new WebCIL container
+format.
+
+
+## Specification
+
+As our starting point we take section II.25.1 "Structure of the
+runtime file format" from ECMA-335 6th Edition.
+
+| |
+|--------|
+| PE Headers |
+| CLI Header |
+| CLI Data |
+| Native Image Sections |
+| |
+
+
+
+A Webcil file follows a similar structure
+
+
+| |
+|--------|
+| Webcil Headers |
+| CLI Header |
+| CLI Data |
+| |
+
+## Webcil Headers
+
+The Webcil headers consist of a Webcil header followed by a sequence of section headers.
+(All multi-byte integers are in little endian format).
+
+### Webcil Header
+
+``` c
+struct WebcilHeader {
+	uint8_t id[2]; // 'W' 'C'
```

Comment:
The magic header should be longer to allow tools like 'file' to identify this file format.
The version should probably be 4 bytes as well (major-minor).

Reply:
yea I can make the magic longer.

Do we really need a major/minor version scheme? I can make it a bit wider, but I don't think we should be very flexible about mismatches. It's like the AOT image format - you have to match the runtime's expected version exactly, or else we fail.

Reply:
Maybe not, but it's better to have it because it would be hard to add it later.

Reply:
Ok, changed the magic to `WbIL` and made the version two 16 bit values.